src/parseFromAnchor.js

```diff
@@ -1,14 +1,44 @@
 'use strict';
 
 /**
+ * Helper function to determine whether the given uri contains port section
+ * @param {string} uri String URI to check
+ */
+function isPortProvidedToURL(uri) {
+	return /:[0-9]+/.test(uri);
```

Comment:
Could this fail for basic auth urls with passwords?

`scheme:[//[user[:password]@]host[:port]][/path][?query][#fragment]`

Reply:
Well spotted @brunobasto! It could if the password section is purely consisted of numbers, but it is still an eventuality that can provide false-positives. I'm fixing it now.